### PR TITLE
Updates to configuration file storage location

### PR DESCRIPTION
**Pull request by ellismg**
This PR makes two changes:

1. When the `config` directive in `Pulumi.yaml` is unset, we now prefer storing configuration in a directory named `.pulumi` next to the `Pulumi.yaml` file. When set, the old behavior is retained (the directive there is combined with the path to the `Pulumi.yaml` file and placed there).

2. When using the Pulumi Service, store configuration in a folder named after the owner of the stack being configured. This means if I have a stack with the same name across organizations, they now have different configuration.

The individual commits explain more about the changes and why we are making them.

These changes do not attempt to "upgrade" the on disk layout. If a configuration file exists at the older location, it is preferred.  I would be open to adding some warnings on notes in this case that you may want to move to the newer scheme, but my goal here was to prevent something like: "A coworker updated their version of the CLI, made some changes and now everyone on older clients is screwed"

### CODE REVIEW

**Comment by ellismg** [comment]
@pgavlin We talked about this in person, at the time you had a question as to if the hidden directory was going to lead to problems where folks wouldn't see the `.pulumi` folder, and hence may forget to check in these files.

I confirmed that `git status` behaves as expected, showing the `.pulumi` folder, and then when you git add things work as expected:

```
[matell@matell tmp.XMGHwQUimR]$ git status
On branch master

No commits yet

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	.gitignore
	.pulumi/
	Pulumi.yaml
	index.ts
	package-lock.json
	package.json
	tsconfig.json

nothing added to commit but untracked files present (use "git add" to track)
[matell@matell tmp.XMGHwQUimR]$ git add .
[matell@matell tmp.XMGHwQUimR]$ git status
On branch master

No commits yet

Changes to be committed:
  (use "git rm --cached <file>..." to unstage)

	new file:   .gitignore
	new file:   .pulumi/ellismg/Pulumi.tmp.XMGHwQUimR-dev.yaml
	new file:   Pulumi.yaml
	new file:   index.ts
	new file:   package-lock.json
	new file:   package.json
	new file:   tsconfig.json

[matell@matell tmp.XMGHwQUimR]$ 
```

This was from a folder where I just did `pulumi new aws-typescript` followed by `git init`.

**Comment by ellismg** [comment]
@lukehoban and I spent some time talking about this and I think we want to hold off on it a little longer.

One thing we talked about is if the existence of the `.pulumi` directory here could function in a way the `Pulumi.yaml` file does today.

At a high level, `Pulumi.yaml` is necessary because it is used as the "root" of the project.  In addition the `name` and `runtime` fields are both necessary today.

If we had a `.pulumi` folder, you could imagine removing the need for an explicit Pulumi.yaml file.  If it didn't exist, we could say things like: "the name of your project is the name of the parent folder", the runtime is determined from a set of dumb heuristics (i.e. we see a package.json next to .pulumi, you are probably a node app).

If these values needed to be overridden, we could do so via a yaml file in the root of the `.pulumi` folder (you could imagine that `Pulumi.yaml`, when needed, would live in the root of `.pulumi`)

The other interesting tension here is issues like https://github.com/pulumi/pulumi/issues/2306 which implies that hand editing these yaml files (instead of managing them with `pulumi config set` is more common (which, of course, in retrospect seems super natural, but I had not really ever imagined doing this). If we expect hand editing these files is super common does that change how we want to handle this?

We probably need to think about requests like https://github.com/pulumi/pulumi/issues/2307 at the same time before we pull the trigger on this.

In short, he and I were both a little about just pushing this in as is, because we are worried it's going to quickly become wrong in some way and yet another thing we'll have to support for a while.
